sw/device/lib/testing/dif/mock_dif_uart_test.cc

```diff
@@ -0,0 +1,228 @@
+// Copyright lowRISC contributors.
```

Comment:
I think these files should live side-by-side with the targets they are testing. I don't think there's any danger of confusion with "actual" DIF libraries, and makes finding the tests much easier.

Google C++ does not strictly require this, but it does tacitly recommend it in a few places.

Reply:
I am happy with `dif_uart_test.cc` and don't really have strong opinion on the naming. One thing I was wondering, whether it is worth calling this directory `mock_testing/mock_tests` instead of `testing`. Just so that anyone who has a look at the directory layout, can see the difference without poking around (we have also sw/device/tests/...). What do you think guys?

Reply:
While I don't have a strong feeling on this, I would prefer having tests and their targets in the same directory (as @mcy suggested).

Reply:
I think I misunderstood what you guys were proposing, do you mean having `dif_uart_test.cc` in `sw/device/lib/dif/` ?

Reply:
@silvestrst yes that is what they are proposing. 

I would like it to be quite clear what is production device code and what is not (in this case: host-based C++ tests), and choosing a different directory seems the clearest way of doing so. At lowrisc we seem to agree that `sw/device/tests/dif/dif_uart_test_mocked.cc` is the best place for this test file, and we may move the existing smoke tests in `sw/device/tests` into a subdirectory that makes it clearer what kind of test they are.

Reply:
> I think I misunderstood what you guys were proposing, do you mean having `dif_uart_test.cc` in `sw/device/lib/dif/` ?

Yes, that was the proposal but I'm OK with putting them in a separate directory for the sake of clarity and consistency (smoke tests already live in a different directory).

Reply:
NOTE: still haven't addressed this!

Reply:
Done.